gitbook/graphics-sprite.md

```diff
@@ -2,204 +2,178 @@
 
-This problem occurs when you write this kind of function: 
-
-```
-sf::Sprite loadSprite(std::string filename)
-{
-    sf::Texture texture;
-    texture.loadFromFile(filename);
-
-    return sf::Sprite(texture);
-} // error: the texture is destroyed here
-```
-
-You must correctly manage the lifetime of your textures and make sure that they live as long as they are used by any sprite. 
+Since transformation functions are common to all CrSFML entities, they are explained in a separate tutorial: [Transforming entities](graphics-transform.md "'Transforming entities' tutorial").
 
 ## The importance of using as few textures as possible
 
-Using as few textures as possible is a good strategy, and the reason is simple: Changing the current texture is an expensive operation for the graphics card. Drawing many sprites that use the same texture will yield the best performance. 
+Using as few textures as possible is a good strategy, and the reason is simple: Changing the current texture is an expensive operation for the graphics card. Drawing many sprites that use the same texture will yield the best performance.
 
-Additionally, using a single texture allows you to group static geometry into a single entity (you can only use one texture per `draw` call), which will be much faster to draw than a set of many entities. Batching static geometry involves other classes and is therefore beyond the scope of this tutorial, for further details see the [vertex array](./graphics-vertex-array.html "Vertex array tutorial") tutorial. 
+Additionally, using a single texture allows you to group static geometry into a single entity (you can only use one texture per `draw` call), which will be much faster to draw than a set of many entities. Batching static geometry involves other classes and is therefore beyond the scope of this tutorial, for further details see the [vertex array](graphics-vertex-array.md "Vertex array tutorial") tutorial.
 
-Try to keep this in mind when you create your animation sheets or your tilesets: Use as little textures as possible. 
+Try to keep this in mind when you create your animation sheets or your tilesets: Use as little textures as possible.
 
-## Using sf::Texture with OpenGL code
+## Using SF::Texture with OpenGL code
 
-If you're using OpenGL rather than the graphics entities of SFML, you can still use [Texture]({{book.api}}/Texture.html) as a wrapper around an OpenGL texture object and use it along with the rest if your OpenGL code. 
+If you're using OpenGL rather than the graphics entities of CrSFML, you can still use [Texture]({{book.api}}/Texture.html) as a wrapper around an OpenGL texture object and use it along with the rest if your OpenGL code.
 
-To bind a [Texture]({{book.api}}/Texture.html) for drawing (basically `glBindTexture`), you call the `bind` static function: 
+To bind a [Texture]({{book.api}}/Texture.html) for drawing (basically `glBindTexture`), you call the `bind` static function:
```

Comment:
`texture.bind` is what's present now. The automatic generator couldn't know that this is supposed to be a class method.

But yes... it's impossible to bind `NULL`, I need to implement it like `SF::Texture.bind(texture)`, `SF::Texture.bind(nil)`.